build.py

```diff
@@ -271,6 +243,34 @@ def install_qsharp_python_package(cwd, wheelhouse, interpreter):
     subprocess.run(command_args, check=True, text=True, cwd=cwd)
 
 
+# If any package fails to install when using a requirements file, the entire
+# process will fail with unpredicatable state of installed packages. To avoid
+# this, we install each package individually from the requirements file.
+#
+# The reason we allow failures is that tooling for integration tests may not
+# be available on all platforms, so we don't want to fail the build if we can't
+# run the tests. The CI will run the tests on the platforms where the tooling
+# is available giving us the confidence that the tests pass on those platforms.
+def install_python_test_requirements(cwd, interpreter, check: bool = True):
+    requirements_file_path = os.path.join(cwd, "test_requirements.txt")
+    with open(requirements_file_path, "r", encoding="utf-8") as f:
+        # Skip empty lines
+        requirements = [line for line in f if line.strip()]
+    for requirement in requirements:
+        command_args = [
+            interpreter,
+            "-m",
+            "pip",
+            "install",
+            requirement,
+            "--only-binary",
+            "qirrunner",
+            "--only-binary",
+            "pyqir",
```

Comment:
nitpick: since I doubt we want any source packages in this part of the build, would it make sense to collapse this to the `:all:` special case?
```suggestion
            "--only-binary",
            ":all:",
```